src/sfloader/fluid_defsfont.c

```diff
@@ -867,6 +868,25 @@ fluid_defpreset_noteon(fluid_defpreset_t *defpreset, fluid_synth_t *synth, int c
     fluid_voice_zone_t *voice_zone;
     fluid_list_t *list;
     fluid_voice_t *voice;
+    int tuned_key;
+
+    /* For detuned channels it might be better to use another key for Soundfont sample selection
+     * giving better approximations for the pitch than the original key.
+     * Example: play key 60 on 6370 Hz => use tuned key 64 for sample selection
+     *
+     * This feature is only enabled for melodic channels.
+     * For drum channels we always select Soundfont samples by key numbers.
+     */
+
+    if(synth->channel[chan]->channel_type == CHANNEL_TYPE_MELODIC)
+    {
+        tuned_key = (int)(fluid_channel_get_key_pitch(synth->channel[chan], key) / 100.0 + 0.5);
+    }
+    else
+    {
+        tuned_key = key;
+    }
+
     int i;
```

Comment:
Move declaration of `int i` up to avoid `warning: ISO C90 forbids mixed declarations and code`